packages/kit/src/core/adapt/prerender/prerender.js

```diff
@@ -229,7 +229,9 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 					if (!is_root_relative(resolved)) continue;
 
 					const parsed = new URL(resolved, 'http://localhost');
-					const pathname = decodeURI(parsed.pathname).replace(config.kit.paths.base, '');
+
+					if (!parsed.pathname.startsWith(config.kit.paths.base)) continue;
```

Comment:
Do we need to decode first before checking this?

Reply:
In a couple of [other](https://github.com/sveltejs/kit/blob/9e482ac43b2d3f903f497ab5c9cc92691aec9906/packages/kit/src/core/preview/index.js#L94) [places](https://github.com/sveltejs/kit/blob/9e482ac43b2d3f903f497ab5c9cc92691aec9906/packages/kit/src/core/dev/plugin.js#L138), it seems to be checked without decoding. `paths.base` doesn't seem to be documented one way or the other (whether it's encoded or decoded); on the other hand, the encoded form isn't canonical, so e.g. even with a "normal"-looking base of `/foo`, someone visiting via `/%66%6f%6f` might not match, even though it's the "same" URL. (The same issue, I would imagine, applies to the other places where it's checked as well - including probably in adapters - and behavior would vary depending on whether whatever is upstream canonicalizes the URL.) Honestly, not sure how much of a problem that is.

Reply:
updated the PR